packages/core/data-transfer/src/strapi/providers/local-destination/index.ts

```diff
@@ -0,0 +1,245 @@
+} from '../../../../types';
+
+import { restore } from './strategies';
+import * as utils from '../../../utils';
+import { ProviderTransferError, ProviderValidationError } from '../../../errors/providers';
+import { assertValidStrapi } from '../../../utils/providers';
+
+export const VALID_CONFLICT_STRATEGIES = ['restore', 'merge'];
+export const DEFAULT_CONFLICT_STRATEGY = 'restore';
+
+export interface ILocalStrapiDestinationProviderOptions {
+  getStrapi(): Strapi.Strapi | Promise<Strapi.Strapi>;
+  autoDestroy?: boolean;
+  restore?: restore.IRestoreOptions;
+  strategy: 'restore' | 'merge';
+}
+
+class LocalStrapiDestinationProvider implements IDestinationProvider {
+  name = 'destination::local-strapi';
+
+  type: ProviderType = 'destination';
+
+  options: ILocalStrapiDestinationProviderOptions;
+
+  strapi?: Strapi.Strapi;
+
+  transaction?: Transaction;
+
+  /**
+   * The entities mapper is used to map old entities to their new IDs
+   */
+  #entitiesMapper: { [type: string]: { [id: number]: number } };
+
+  constructor(options: ILocalStrapiDestinationProviderOptions) {
+    this.options = options;
+    this.#entitiesMapper = {};
+  }
+
+  async bootstrap(): Promise<void> {
+    this.#validateOptions();
+    this.strapi = await this.options.getStrapi();
+
+    this.transaction = utils.transaction.createTransaction(this.strapi);
+  }
+
+  async close(): Promise<void> {
+    const { autoDestroy } = this.options;
+    this.transaction?.end();
+
+    // Basically `!== false` but more deterministic
+    if (autoDestroy === undefined || autoDestroy === true) {
+      await this.strapi?.destroy();
+    }
+  }
+
+  #validateOptions() {
+    if (!VALID_CONFLICT_STRATEGIES.includes(this.options.strategy)) {
+      throw new ProviderValidationError(`Invalid strategy ${this.options.strategy}`, {
+        check: 'strategy',
+        strategy: this.options.strategy,
+        validStrategies: VALID_CONFLICT_STRATEGIES,
+      });
+    }
+  }
+
+  async #deleteAll() {
+    assertValidStrapi(this.strapi);
+    return restore.deleteRecords(this.strapi, this.options.restore);
+  }
+
+  async rollback(e: Error): Promise<void> {
+    await this.transaction?.rollback();
+  }
+
+  async beforeTransfer() {
+    if (!this.strapi) {
+      throw new Error('Strapi instance not found');
+    }
+
+    await this.transaction?.attach(async () => {
+      try {
+        if (this.options.strategy === 'restore') {
+          await this.#deleteAll();
+        }
+      } catch (error) {
+        throw new Error(`restore failed ${error}`);
+      }
+    });
+  }
+
+  getMetadata(): IMetadata {
+    const strapiVersion = strapi.config.get('info.strapi');
+    const createdAt = new Date().toISOString();
+
+    const plugins = Object.keys(strapi.plugins);
+
+    return {
+      createdAt,
+      strapi: {
+        version: strapiVersion,
+        plugins: plugins.map((name) => ({
+          name,
+          // TODO: Get the plugin actual version when it'll be available
```

Comment:
@Convly Could this potentially cause problems in the future? Should we remove the version from the plugin fields if we don't have it? Or is this fine as-is for now?